elide-async/src/main/java/com/yahoo/elide/async/models/AsyncQuery.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.models;
+
+import java.security.Principal;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import javax.persistence.Transient;
+
+import com.yahoo.elide.annotation.Include;
+import com.yahoo.elide.annotation.OnCreatePostCommit;
+import com.yahoo.elide.annotation.ReadPermission;
+import com.yahoo.elide.annotation.UpdatePermission;
+import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.core.RequestScope;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Model for Async Query
+ */
+@Entity
+@Include(type = "query", rootLevel = true)
+@ReadPermission(expression = "Principal is Owner")
+@UpdatePermission(expression = "Prefab.Role.None")
+@Slf4j
+public class AsyncQuery implements PrincipalOwned {
```

Comment:
All of the Elide models should implement hashcode & equals.  There are some best practices around how to make this work with ORMs.  You can look at BaseId in Elide to get one of the approaches.

Reply:
We are adding @Data for AsyncQuery.java which internally includes @EqualsAndHashCode. Do we need to add these separately for the models?

Reply:
`@Data` should be sufficient.

Reply:
@Data is not sufficient for use with ORMs.